networkit/cpp/bipartite/test/BFSBipartiteCheckGTest.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+ * BFSBipartiteCheckGTest.cpp
+ *
+ *  Created on: 09.02.2025
+ *      Author: Andreas Scharf (andreas.b.scharf@gmail.com)
+ */
+
+#include <gtest/gtest.h>
+#include <networkit/bipartite/BFSBipartiteCheck.hpp>
+#include <networkit/io/METISGraphReader.hpp>
+namespace NetworKit {
+
+class BFSBipartiteCheckGTest : public testing::Test {
+public:
+    static constexpr int maxNumberOfNodes{10};
+    Graph binaryTreeGraph(count numNodes) {
+        Graph graph(numNodes, true, false, true);
+        for (count i = 0; i < numNodes; ++i) {
+            count leftChild = 2 * i + 1;
+            count rightChild = 2 * i + 2;
+            if (leftChild < numNodes) {
+                graph.addEdge(i, leftChild, static_cast<double>(i));
+            }
+            if (rightChild < numNodes) {
+                graph.addEdge(i, rightChild, static_cast<double>(i));
+            }
+        }
+        return graph;
+    }
+
+    Graph completeGraph(count numNodes) {
+        Graph graph(numNodes, true);
+
+        for (count i = 0; i < numNodes; ++i) {
+            for (count j = i + 1; j < numNodes; ++j) {
+                graph.addEdge(i, j, static_cast<double>(j * (j + 1)));
+            }
+        }
+        return graph;
+    }
+};
+
+TEST_F(BFSBipartiteCheckGTest, testDirectedGraphThrows) {
+    Graph graph(0, false, true, false);
+    try {
+        BFSBipartiteCheck test(graph);
+        FAIL() << "Expected std::runtime_error";
+    } catch (const std::runtime_error &e) {
+        EXPECT_STREQ(e.what(), "The graph is not an undirected graph.");
+    } catch (...) {
+        FAIL() << "Expected std::runtime_error but got a different exception.";
+    }
+}
+
+TEST_F(BFSBipartiteCheckGTest, testIsBipartiteThrowsIfRunIsNotCalled) {
+    Graph graph{};
+    BFSBipartiteCheck test(graph);
+    try {
+        test.isBipartite();
+        FAIL() << "Expected std::runtime_error";
+    } catch (const std::runtime_error &e) {
+        EXPECT_STREQ(e.what(), "Error, run must be called first");
+    } catch (...) {
+        FAIL() << "Expected std::runtime_error but got a different exception.";
+    }
+}
+
+TEST_F(BFSBipartiteCheckGTest, testBipartiteEmptyGraph) {
+    Graph graph{};
+    BFSBipartiteCheck test(graph);
+    test.run();
+    EXPECT_TRUE(test.isBipartite());
+}
+
+TEST_F(BFSBipartiteCheckGTest, testBipartiteSingleNodesGraphs) {
+    for (count i = 0; i < maxNumberOfNodes; ++i) {
```

Comment:
Here you are repeating the same test on a graph with one node `maxNumberOfNodes` times, one is enough.